pyshelf/search/update_manager.py

```diff
@@ -0,0 +1,73 @@
+from pyshelf.search.metadata import Metadata
+
+
+class UpdateManager(object):
+    def __init__(self, logger):
+        self.logger = logger
+
+    def bulk_update(self, data):
+        """
+            This provides bulk updating functionality. It has the ability to update multiple documents.
+
+            Args:
+                data(dict): This contains metadata and the associated document key. Example below:
+
+            Example of data format:
+            {
+                "key_of_doc": {
+                    "artifactPath": {
+                        "name": "artifactPath",
+                        "value": "/wibble/wobble",
+                        "immutable": False
+                    },
+                    ....
+                },
+                ....
+            }
+        """
+        for key, val in data.iteritems():
+            self.update(key, val)
+
+    def update(self, key, metadata):
+        """
+            Updates the metadata in the ElasticSearch collection denoted by the supplied unique key.
+
+            Args:
+                key(string): Unique key that is associated with the metadata document to update.
+                metadata(dict): Updated metadata to store in ElasticSearch.
+        """
+        self.logger.debug("Attempting update of metadata: {0} in ES".format(key))
+        meta_doc = self.get_metadata(key)
+        meta_doc.update_all(metadata)
+        self.logger.debug("Updated metadata document {0} in ES".format(key))
+
+    def update_item(self, key, item):
+        """
+           Updates metadata item in the ElasticSearch collection.
+
+            Args:
+                key(string): Unique key that is associated with the metadata document to update.
+                item(dict): Updated metadata to store in ElasticSearch.
+        """
+        self.logger.debug("Attempting to update metadata {0} in ES".format(key))
+        meta_doc = self.get_metadata(key)
+        meta_doc.update_item(item)
+        self.logger.debug("Updated metadata {0} in ES".format(key))
+
+    def get_metadata(self, key):
```

Comment:
Can you make this "private"?  Its not something that should get used outside the search layer correct?  I wouldn't want the user to deal with a pyshelf.search.metadata.Metadata